packages/xo-web/src/common/xo/index.js

```diff
@@ -821,21 +822,87 @@ export const setRemoteSyslogHost = (host, syslogDestination) =>
 export const setRemoteSyslogHosts = (hosts, syslogDestination) =>
   Promise.all(map(hosts, host => setRemoteSyslogHost(host, syslogDestination)))
 
-export const restartHost = (host, force = false, suspendResidentVms = false) =>
+export const restartHost = (
+  host,
+  force = false,
+  suspendResidentVms = false,
+  bypassBlockedSuspend = false,
+  bypassCurrentVmCheck = false
+) =>
   confirm({
     title: _('restartHostModalTitle'),
     body: _('restartHostModalMessage'),
   }).then(
     () =>
-      _call('host.restart', { id: resolveId(host), force, suspendResidentVms })
+      _call('host.restart', {
+        id: resolveId(host),
+        force,
+        suspendResidentVms,
+        bypassBlockedSuspend,
+        bypassCurrentVmCheck,
+      })
+        .catch(async error => {
+          if (
+            suspendResidentVms &&
+            !bypassBlockedSuspend &&
+            incorrectState.is(error, {
+              object: 'suspendBlocked',
+            })
+          ) {
+            await confirm({
+              body: (
+                <p>
+                  <Icon icon='alarm' /> {_('forceSmartRebootHost', { nVms: error.data.actual.length })}
+                </p>
+              ),
+              title: _('restartHostModalTitle'),
+            })
+            bypassBlockedSuspend = true
+            return _call('host.restart', {
+              id: resolveId(host),
+              force,
+              suspendResidentVms,
+              bypassBlockedSuspend,
+            })
+          }
+          throw error
+        })
         .catch(async error => {
           if (
-            forbiddenOperation.is(error, {
+            error !== undefined &&
+            operationFailed.is(error, {
+              code: 'xoaOnHost',
+            })
+          ) {
+            await confirm({
+              body: (
+                <p>
+                  <Icon icon='alarm' /> {_('smartRebootBypassCurrentVmCheck')}
+                </p>
+              ),
+              title: _('restartHostModalTitle'),
+            })
+            bypassCurrentVmCheck = true
+
+            return _call('host.restart', {
+              id: resolveId(host),
+              force,
+              suspendResidentVms,
+              bypassBlockedSuspend,
+              bypassCurrentVmCheck,
+            })
+          }
+          throw error
+        })
+        .catch(async error => {
+          if (
+            error !== undefined &&
+            (forbiddenOperation.is(error, {
               reason: `A backup may run on the pool: ${host.$poolId}`,
             }) ||
-            forbiddenOperation.is(error, {
-              reason: `A backup is running on the pool: ${host.$poolId}`,
-            })
+              forbiddenOperation.is(error, {
+                reason: `A backup is running on the pool: ${host.$poolId}`,
+              }))
```

Comment:
Since it's getting quite complex, I'm thinking we might be able to simplify it and avoid rethrowing the error every time with something like this:
```js
function restart(opts) {
  await confirm('restart?')
  await _restart(opts)
}

function _restart(opts) {
  try {
    await call('host.restart', opts)
  } catch (err) {
    if (cantSuspend(err)) {
      await confirm('Cannot suspend. Force?')
      return _restart({ ...opts, bypassBlockedSuspend: true })  
    }

    if (xoaOnHost(err)) {
      await confirm('XOA will restart. Force?')
      return _restart({ ...opts, bypassCurrentVmCheck: true })  
    }

    if (backupIsRunning(err)) {
      await confirm('Backup is running. Force?')
      return _restart({ ...opts, bypassBackupCheck: true })  
    }

    if (noHostAvailable(err)) {
      alert('No host available')
    }

    throw err
  }
}
```
It also adds the small advantage of retesting all the possible errors every time in case something changed between 2 calls and the first error type of the list only gets triggered on the 2nd call.

What do you think?